### PR TITLE
new(modern_bpf): add support for futex syscall

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

/area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:
Add support for futex syscall in modern_bpf

**Which issue(s) this PR fixes**:

Part of: #723 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/futex.bpf.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	/* Parameter 1: addr (type: PT_UINT64) */
+	u64 addr = extract__syscall_argument(regs, 0);
+	ringbuf__store_u64(&ringbuf, addr);
+
+	/* Parameter 2: op (type: PT_ENUMFLAGS16) */
+	s32 op = (s32)extract__syscall_argument(regs, 1);
+	ringbuf__store_u16(&ringbuf, futex_op_to_scap((unsigned long)op));
+
+	/* Parameter 3: val (type: PT_UINT64) */
+	u64 val = extract__syscall_argument(regs, 2);
```
Comment (loresuso) [inline driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/futex.bpf.c:36]: another example of wasting bytes in our buffers here.. val is an uint32_t and we are pushing 8 bytes :( 
as far as I can see, we sometimes optimize other pieces of the libs but our events are not

**Review by Andreagit97 — APPROVED** [review]
Thank you for that!
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**